crates/ruff_python_parser/src/parser/progress.rs

```diff
@@ -1,18 +1,27 @@
 use crate::parser::Parser;
-use crate::TokenKind;
-use ruff_text_size::TextSize;
+
+#[derive(Copy, Clone, Debug, Default, PartialEq, Eq)]
+pub(super) struct TokenId(u32);
+
+impl TokenId {
+    /// Increments the value of the token ID.
+    pub(super) fn increment(&mut self) {
+        // SAFETY: We don't support files larger than 4GB, so this should never overflow.
+        self.0 = self.0.checked_add(1).expect("TokenId overflow");
```

Comment:
Hmm, I'm confused. I'm pretty sure that I wrote a comment when I reviewed this PR but it seems gone. 

Anyway, it's a Nit but I don't think it's necessary to use `checked_add` here because we only need to know if it's still the same token. True, it would be possible that `ParserProgress` doesn't recognize when an iteration exactly processed `u32::MAX` items but this isn't a really issue because of what you outline in the SAFETY comment

```suggestion
self.0 = self.0.wrapping_add(1)
```

Reply:
Oh right. I need more coffee :)